src/java/org/apache/cassandra/cql3/statements/schema/CopyTableStatement.java

```diff
@@ -0,0 +1,175 @@
+    }
+
+    @Override
+    public void validate(ClientState state)
+    {
+        super.validate(state);
+    }
+
+    @Override
+    public Keyspaces apply(ClusterMetadata metadata)
+    {
+        Keyspaces schema = metadata.schema.getKeyspaces();
+        KeyspaceMetadata sourceKeyspaceMeta = schema.getNullable(sourceKeyspace);
+        TableMetadata sourceTableMeta = sourceKeyspaceMeta.getTableNullable(sourceTableName);
+
+        if (null == sourceKeyspaceMeta)
+            throw ire("Source Keyspace '%s' doesn't exist", sourceKeyspace);
+
+        if (null == sourceTableMeta)
+        {
+            throw ire("Souce Table '%s'.'%s' doesn't exist", sourceKeyspace, sourceTableName);
+        }
+
+        if (sourceTableMeta.isIndex())
+            throw ire("Cannot use CTREATE TABLE LIKE on a index table '%s'.'%s'.", sourceKeyspace, sourceTableName);
+
+        if (sourceTableMeta.isView())
+            throw ire("Cannot use CTREATE TABLE LIKE on a materialized view '%s'.'%s'.", sourceKeyspace, sourceTableName);
+
+        KeyspaceMetadata targetKeyspaceMeta = schema.getNullable(targetKeyspace);
+        if (null == targetKeyspaceMeta)
+            throw ire("Target Keyspace '%s' doesn't exist", targetKeyspace);
+
+        if (targetKeyspaceMeta.hasTable(targetTableName))
+        {
+            throw new AlreadyExistsException(targetKeyspace, targetTableName);
+        }
+
+        // todo support udt for differenet ks latter
+        if (!sourceKeyspace.equalsIgnoreCase(targetKeyspace) && !sourceKeyspaceMeta.types.isEmpty())
+        {
+            throw ire("Cannot use CTREATE TABLE LIKE across different keyspace when source table have UDTs.");
+        }
+
+        String sourceCQLString = sourceTableMeta.toCqlString(false, false, true, false);
+        // add all user functions to be able to give a good error message to the user if the alter references
+        // a function from another keyspace
+        UserFunctions.Builder ufBuilder = UserFunctions.builder().add();
+        for (KeyspaceMetadata ksm : schema)
+            ufBuilder.add(ksm.userFunctions);
+
+        //todo support table params' setting in the future
+        TableMetadata.Builder targetBuilder = CreateTableStatement.parse(sourceCQLString,
+                                                                         targetKeyspace,
+                                                                         targetTableName,
+                                                                         sourceKeyspaceMeta.types,
+                                                                         ufBuilder.build())
+                                                                  .indexes(Indexes.none())
+                                                                  .triggers(Triggers.none());
+
+        TableMetadata table = targetBuilder.id(TableId.get(metadata)).build();
```

Comment:
you are not doing it like it is in CreateTableStatement

````
        // We do not want to set table ID here just yet, since we are using CQL for serialising a fully expanded CREATE TABLE statement.
        this.expandedCql = builder.build().toCqlString(false, attrs.hasProperty(TableAttributes.ID), ifNotExists);

        if (!attrs.hasProperty(TableAttributes.ID))
            builder.id(TableId.get(metadata));
        TableMetadata table = builder.build();
        table.validate();
````

you dont deal with `expandedCql`

Reply:
you also do not override what is in CreateTableStatement

````

    @Override
    public String cql()
    {
        if (expandedCql != null)
            return expandedCql;
        return super.cql();
    }

````

Reply:
@Maxwell-Guo do you think it is possible to introduce the definition in table attributes for copy statement in this PR? If we know how to do that, as you found the way to do it, it does not make a lot of sense to have it in a separate PR. I would try to do as much as possible in one PR and we do not need to return to that in the future. 

Feel free to expand this PR so the definition of table attributes and the overriding of them is supported as well, based on the approach I showed here 

https://github.com/apache/cassandra/pull/3722/commits/9b36cfecf7612271cbc0dde9cd16ada6f6756919

Reply:
Just do it in this PR, no "new" PR necessary. Just update this.